docs/standards/smirnoff.md

```diff
@@ -638,6 +638,16 @@ To specify the constraint distance, or constrain two atoms that are not directly
   <!-- constrain water H...H, calculating equilibrium length from H-O-H equilibrium angle and H-O equilibrium bond lengths -->
   <Constraint smirks="[#1:1]-[#8X2H2]-[#1:2]" distance="1.8532*angstrom"/>
 </Constraints>
+
+If a pair of atoms is subject to parameters in both the `<Bonds>` and `<Constraints>` sections and
+the `<Constraint>` parameter specifies a `distance`, the bond is constrained with the `distance`
+value specified by the corresponding `<Constraints>` record, not the `distance` value specified by
+the `<Bonds>` record.
+
+If a pair of atoms is subject to parameters in both the `<Bonds>` and `<Constraints>` sections and
+the `<Constraint>` parameter does not specify a `distance`, the bond is constrained to a distance of
+the `length` value in the `<Bond>` parameter.
```

Comment:
This could maybe be consolidated to:

```suggestion
If a constraint is applied across a bond between two atoms, then the length of that bond will be constrained to:

* the value of the `distance` attribute of the `<Constraint>` parameter _if one is specified_, *otherwise*
* the value of the `length` attribute of the `<Bond>` parameter that is matched by that bond
```
We then probably want to clarify the edge case of:

```
If the constraint does *not* specify a `distance`, and is applied to two atoms that either aren't bonded or which do not have an associated bond parameter, then an exception should be raised.
```